server/api/utils/projects/follower.py

```diff
@@ -66,32 +71,24 @@ def initialize(self):
         )
         self._synced_until_datetime = None
         # run one sync to start off on the right foot and fill out the cache but don't fail initialization on it
-        try:
-            # Basically the delete operation in our projects mechanism is fully consistent, meaning the leader won't
-            # remove the project from its persistency (the source of truth) until it was successfully removed from all
-            # followers. Therefore, when syncing projects from the leader, we don't need to search for the deletions
-            # that may happen without us knowing about it (therefore full_sync by default is false). When we
-            # introduced the chief/worker mechanism, we needed to change the follower to keep its projects in the DB
-            # instead of in cache. On the switch, since we were using cache and the projects table in the DB was not
-            # maintained, we know we may have projects that shouldn't be there anymore, ideally we would have trigger
-            # the full sync only once on the switch, but since we don't have a good heuristic to identify the switch
-            # we're doing a full_sync on every initialization
-            full_sync = (
-                mlrun.mlconf.httpdb.clusterization.role
-                == mlrun.common.schemas.ClusterizationRole.chief
-            )
-            self._sync_projects(full_sync=full_sync)
-        except Exception as exc:
-            logger.warning(
-                "Initial projects sync failed",
-                exc=err_to_str(exc),
-                traceback=traceback.format_exc(),
-            )
-        self._start_periodic_sync()
+        if self._chief:
+            try:
+                # full_sync=True was a temporary measure to handle the move of mlrun from single instance to
+                # chief-worker model.
+                # TODO: remove full_sync=True in 1.7.0 if no issues arise
```

Comment:
we do full_sync only on initialize and not on periodic so you will still need it here even of you don't sync in workers

Reply:
discussed offline